libgadget/cooling_rates.c

```diff
@@ -174,12 +184,12 @@ load_treecool(const char * TreeCoolFile)
                 continue;
             Gamma_log1z[i] = atof(retval);
             /*Get the rest*/
-            Gamma_HI.ydata[i] = log10(atof(strtok(NULL, " \t")));
-            Gamma_HeI.ydata[i] = log10(atof(strtok(NULL, " \t")));
-            Gamma_HeII.ydata[i] = log10(atof(strtok(NULL, " \t")));
-            Eps_HI.ydata[i] = log10(atof(strtok(NULL, " \t")));
-            Eps_HeI.ydata[i] = log10(atof(strtok(NULL, " \t")));
-            Eps_HeII.ydata[i] = log10(atof(strtok(NULL, " \t")));
+            Gamma_HI.ydata[i]   = load_tree_value();
```

Comment:
Better to use a locally defined macro than hiding strtok, which is a weird function that uses a hidden state. We may forget load_tree_value also has that strange behaviour sooner or later.

Reply:
Even better to not use the weird strtok function: the last commit has strtok_r